### PR TITLE
Feat: Add import exam command

### LINKED ISSUE #229: Implement Import Exam

### PR DESCRIPTION
**Pull request by Zer0Legion**
fixes #229

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ImportExamCommand.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+            AddScoreCommandParser addScoreCommandParser = new AddScoreCommandParser();
+            AddScoreCommand addScoreCommand = addScoreCommandParser.parse(PREFIX_SCORE + grade.toString());
+            addScoreCommand.execute(model);
+        } catch (ParseException e) {
+            throw new CommandException(e.getMessage());
+        } catch (CommandException e) {
+            // Already checked if person exists. Hence, this should not happen.
+            assert model.getFilteredPersonList().size() > 0;
+            String email = String.valueOf(model.getFilteredPersonList().get(0).getEmail());
+            addToErrorReport(email, e.getMessage());
```
Comment (delishad21) [inline src/main/java/seedu/address/logic/commands/ImportExamCommand.java:264]: Good Job on handling failure on addscore

File: src/main/java/seedu/address/logic/parser/ImportExamCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public ImportExamCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_IMPORT);
+        if (!isPrefixPresent(
+                argMultimap)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ImportExamCommand.MESSAGE_USAGE));
+        }
+
```
Comment (delishad21) [inline src/main/java/seedu/address/logic/parser/ImportExamCommandParser.java:32]: !!! Should check if there are duplicate entries for "i/". Can do this with Argmultimap.verifyNoDuplicatePrefixesFor(PREFIX_IMPORT)

**Review by delishad21 — COMMENTED** [review]
Clean and easy to understand implementation! Well done.

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public class ImportCommand extends Command {
     public static final String MESSAGE_NOT_IMPLEMENTED_YET =
             "Remark command not implemented yet";
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports contacts from specified filepath."
-            + " Must be an absolute filePath\n"
+            + " Must be an absolute CSV file path\n"
```
Comment (Pughal77) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:43]: Thanks for editting this!

File: src/main/java/seedu/address/logic/parser/ImportExamCommandParser.java (hunk 2 above)
Comment (Pughal77) [reply src/main/java/seedu/address/logic/parser/ImportExamCommandParser.java:32]: Good catch!

File: src/main/java/seedu/address/logic/commands/ImportExamCommand.java [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+     * @return the error report
+     */
+    static String generateErrorReport() {
+        String errorReportString = errorReport.toString();
+        if (errorReportString.isBlank()) {
+            return "";
+        } else {
+            return PREFIX_ERROR_REPORT + errorReportString;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Pughal77) [reply src/main/java/seedu/address/logic/commands/ImportExamCommand.java:79]: Yes! Perhaps I should also add this in import command to handle errors thrown in importing of contacts!

**Review by Pughal77 — COMMENTED** [review]
Good addition to our product now we can import exams and have it reflected in our GUI!

**Pull request merged**